### PR TITLE
[Hotfix][Zeta] Fix job crash when schema changing on multiple parallel readers

### PR DESCRIPTION
**Pull request by dailai**
### Purpose of this pull request

Fix the close task exception during Full incremental switching.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by dailai** [comment]
When I used this pr(#6929) to test the schema change using multi-parallel tasks, I found that the schema change during the full incremental switch of cdc tasks would result in the following exceptions. Link #6526 #6389 
截屏2024-06-13 10 58 00

**Comment by dailai** [comment]
After the modification, the tests are normal for many times. You can also see the effect in logs.

截屏2024-06-13 11 00 21

**Comment by hailin0** [comment]
The PR that caused this issue
https://github.com/apache/seatunnel/pull/6389

Fix suggestion：remove this `IllegalStateException` and add logs
image

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/SourceFlowLifeCycle.java [hunk 1]
```diff
@@ -310,6 +310,13 @@ public void triggerBarrier(Barrier barrier) throws Exception {
                         "lock checkpoint[{}] waiting for complete..., phase: [{}]",
                         barrier.getId(),
                         schemaChangePhase.get().getPhase());
+            } else if (prepareClose) {
+                TaskLocation taskLocation = runningTask.getTaskLocation();
+                log.info(
+                        "Task[{}]({}/{}) prepare close",
+                        taskLocation.getTaskID(),
+                        taskLocation.getPipelineId(),
+                        taskLocation.getJobId());
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/SourceFlowLifeCycle.java]:
```suggestion
            } else {
                log.debug("Ignore schema-change checkpoint[{}], phase: [{}]", barrier.getId(), checkpointType);
```

Comment (dailai) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/SourceFlowLifeCycle.java]: done

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/SourceFlowLifeCycle.java [hunk 2]
```diff
@@ -311,10 +311,10 @@ public void triggerBarrier(Barrier barrier) throws Exception {
                         barrier.getId(),
                         schemaChangePhase.get().getPhase());
             } else {
-                throw new IllegalStateException(
-                        String.format(
-                                "schema-change checkpoint[%s] and phase[%s] is not matched",
-                                barrier.getId(), checkpointType));
+                log.debug(
+                        "Ignore schema-change checkpoint[{}], phase: [{}]",
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/SourceFlowLifeCycle.java]:
```suggestion
                        "Ignore schema-change checkpoint[{}] on idle task, phase: [{}]",
```

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**